### PR TITLE
APPSRE-7526: RDS backup best practices

### PR DESCRIPTION
**Pull request by bhushanthakur93**
Ensure AWS RDS instances abide by best practices as per https://gitlab.cee.redhat.com/service/app-interface/-/blob/master/docs/app-sre/design-docs/aws-rds-backup-config-best-practices.md.

The integration terraform-resources fails if the database attributes does not comply with necessary attributes based on data_classification and loss_impact.

The exception thrown is of the following format

```
          reconcile.utils.cloud_resource_best_practice.aws_rds.RDSBestPracticesNotComplied: AWS RDS instance my-db does not comply with best practices
          Expected field delete_automated_backups with value "False", found "True" 
          Expected field skip_final_snapshot with value "False", found "True"
```
Schema: https://github.com/app-sre/qontract-schemas/pull/439/files

Testing:

- (Happy path, no exceptions) Create a db with required data_classification and loss_impact, and set the parameters accordingly. Run the terraform-resources integration in dry-run and validate it does not error out.
- (Unhappy path, expect exception! ) Create a db with required data_classification and loss_impact, and set the parameters which are not in line with best practices ( see test case for example ). Run the terraform-resources integration in dry-run and validate exception is thrown.

### CODE REVIEW

File: reconcile/utils/cloud_resource_best_practice/aws_rds.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+from terrascript.resource import aws_db_instance
+
+delete_automated_backups = "delete_automated_backups"
+skip_final_snapshot = "skip_final_snapshot"
+backup_retention_period = "backup_retention_period"
+
+
+# Keeping the exception specific now, we can build a generic exception and refactor later
+# as more cases arise in the future.
+class RDSBestPracticesNotComplied(Exception):
```
Comment (steveteahan) [inline reconcile/utils/cloud_resource_best_practice/aws_rds.py:12]: nit: `RDSResourceComplianceError`? A bit more [pythonic](https://peps.python.org/pep-0008/#exception-names), but only a nit.

File: reconcile/utils/cloud_resource_best_practice/aws_rds.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    (backup_retention_period, 7, OPERATOR_GREATER_THAN_OR_EQUAL_TO),
+]
+
+__loss_impact_medium = [
+    (skip_final_snapshot, False, OPERATOR_EQUAL_TO),
+    (backup_retention_period, 7, OPERATOR_GREATER_THAN_OR_EQUAL_TO),
+]
+__loss_impact_low = [(backup_retention_period, 3, OPERATOR_GREATER_THAN_OR_EQUAL_TO)]
+
+__loss_impact_none = [(backup_retention_period, 0, OPERATOR_GREATER_THAN_OR_EQUAL_TO)]
```
[the remark this answers is not in the document]
Comment (bhushanthakur93) [reply reconcile/utils/cloud_resource_best_practice/aws_rds.py:45]: Nice! This is exactly what I needed. I'll look into this.

[the remark this answers is not in the document]
Comment (bhushanthakur93) [reply reconcile/utils/cloud_resource_best_practice/aws_rds.py]: This is addressed now, PTAL.

File: reconcile/utils/cloud_resource_best_practice/aws_rds.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+                )
+
+    if len(rds_fields_not_complied) > 0:
+        raise RDSBestPracticesNotComplied(
+            rds_fields_not_complied, aws_db_instance._name
+        )
+
+
+def _verify_loss_impact(aws_db_instance: aws_db_instance, loss_impact: str) -> None:
+    if loss_impact == "high":
```
[the remark this answers is not in the document]
Comment (bhushanthakur93) [reply reconcile/utils/cloud_resource_best_practice/aws_rds.py]: Addressed. PTAL at new changes.

**Review by jfchevrette — APPROVED** [review]
LGTM!

**Pull request merged**